include/sensors/firmware/sensor_hardware.hpp

```diff
@@ -11,11 +13,30 @@ class SensorHardware : public SensorHardwareBase {
         : hardware(hardware) {}
     auto set_sync() -> void override { gpio::set(hardware.sync_out); }
     auto reset_sync() -> void override { gpio::reset(hardware.sync_out); }
-    auto check_data_ready() -> bool override {
-        return gpio::is_set(hardware.data_ready);
+
+    auto data_ready() -> void {
```

Comment:
If we expect to have different callback functions at some point (either for multiple pressure sensors, or the environment sensor) then we should maybe have these be key value pairs where the key is the gpio pin. Otherwise, we could end up calling the wrong callback function. It works now because we only have 1 callback function here.

Reply:
If you don't want to address it now, just put a very big TODO here about it.